eventsourcing/src/main/java/org/axonframework/eventsourcing/AbstractSnapshotter.java

```diff
@@ -64,14 +70,23 @@ protected AbstractSnapshotter(Builder builder) {
 
     @Override
     public void scheduleSnapshot(Class<?> aggregateType, String aggregateIdentifier) {
-        if (CurrentUnitOfWork.isStarted()) {
+        if (CurrentUnitOfWork.isStarted() && CurrentUnitOfWork.get().phase().isBefore(UnitOfWork.Phase.COMMIT)) {
             CurrentUnitOfWork.get().afterCommit(u -> doScheduleSnapshot(aggregateType, aggregateIdentifier));
         } else {
             doScheduleSnapshot(aggregateType, aggregateIdentifier);
         }
     }
 
     private void doScheduleSnapshot(Class<?> aggregateType, String aggregateIdentifier) {
+        if (CurrentUnitOfWork.isStarted()) {
```

Comment:
This blocks multiple invocations from the same unit of work, but will not prevent different UoW to trigger the creation of a snapshot....

Reply:
That is correct. However, is it possible to have several UoWs triggering a snapshot on the same Aggregate? (Aggregate is locked while UoW cleans up).